packages/terra-signature/src/Signature.jsx

```diff
@@ -0,0 +1,204 @@
+
+    const rect = canvas.getBoundingClientRect();
+    this.addLine(event.pageX - rect.left, event.pageY - rect.top, false);
+
+    this.draw();
+  }
+
+  mouseUp(event) {
+    this.setState({ painting: false });
+
+    if (this.props.onChange) {
+      this.props.onChange(event, this.state.lineSegments, this.state.lineSegments[this.state.lineSegments.length - 1]);
+    }
+  }
+
+  mouseXY(event) {
+    const canvas = this.node;
+
+    if (this.state.painting) {
+      const rect = canvas.getBoundingClientRect();
+      this.addLine(event.pageX - rect.left, event.pageY - rect.top, true);
+
+      this.draw();
+    }
+  }
+
+  addLine(x, y, dragging) {
+    let newSegment;
+
+    if (dragging) {
+      const lastLineSegment = this.state.lineSegments[this.state.lineSegments.length - 1];
+      newSegment = { x1: lastLineSegment.x2, y1: lastLineSegment.y2, x2: x, y2: y };
+    } else {
+      newSegment = { x1: x, y1: y, x2: x, y2: y };
+    }
+
+    // Record new line segment
+    this.setState({ lineSegments: [...this.state.lineSegments, newSegment] });
+  }
+
+
+  draw() {
+    const canvas = this.node;
+    const context = canvas.getContext('2d');
+
+    if (this.state.lineSegments.length > 0) {
+      const lastLineSegment = this.state.lineSegments[this.state.lineSegments.length - 1];
+
+      context.lineJoin = 'round';
+
+      context.beginPath();
+      context.moveTo(lastLineSegment.x1, lastLineSegment.y1);
+      context.lineTo(lastLineSegment.x2, lastLineSegment.y2);
+
+      context.stroke();
+    }
+  }
+
+  drawSignature(lineSegments) {
+    const canvas = this.node;
+    const context = canvas.getContext('2d');
+
+    context.lineJoin = 'round';
+    context.lineWidth = this.props.lineWidth;
+
+    // clear canvas
+    context.clearRect(0, 0, canvas.getBoundingClientRect().width, canvas.getBoundingClientRect().height);
+
+    // iterate and draw all recorded line segments
+    const segmentCount = lineSegments.length;
+    for (let i = 0; i < segmentCount; i += 1) {
+      context.beginPath();
+      context.moveTo(lineSegments[i].x1, lineSegments[i].y1);
+      context.lineTo(lineSegments[i].x2, lineSegments[i].y2);
+
+      context.stroke();
+    }
+  }
+
+  clearSignature() {
+    this.setState({ lineSegments: [] });
+
+    const canvas = this.node;
+    const context = canvas.getContext('2d');
+
+    context.clearRect(0, 0, canvas.getBoundingClientRect().width, canvas.getBoundingClientRect().height);
+  }
+
+  updateDimensions() {
+    const canvas = this.node;
+
+    this.node.width = canvas.getBoundingClientRect().width;
+    this.node.height = canvas.getBoundingClientRect().height;
+
+    this.drawSignature(this.state.lineSegments);
+  }
+
+  render() {
+    return (
+      <canvas className="terra-Signature" ref={(node) => { this.node = node; }} />
+    );
+  }
+}
+
+Signature.propTypes = propTypes;
+Signature.defaultProps = defaultProps;
+Signature.Width = LineWidths;
+
+export default Signature;
```

Comment:
I don't know how common the use-case is but I would test at least two Signature Components on the screen at the same time and see if the events are bound appropriately and allow the signatures to function independently.